test/integration/process-return-settlement.test.js

```diff
@@ -0,0 +1,116 @@
+const db = require('../../app/data')
+const schemes = require('../../app/constants/schemes')
+
+const processReturnSettlement = require('../../app/inbound/process-return-settlement')
+
+let settlement
+let paymentRequest
+
+describe('process submit payment request', () => {
+  beforeAll(async () => {
+    await db.sequelize.truncate({
+      cascade: true,
+      restartIdentity: true
+    })
+  })
+
+  beforeEach(async () => {
+    jest.useFakeTimers().setSystemTime(new Date(2022, 7, 5, 12, 0, 0, 0))
+    settlement = JSON.parse(JSON.stringify(require('../mock-settlement')))
+    paymentRequest = JSON.parse(JSON.stringify(require('../mock-payment-request').processingPaymentRequest))
+
+    await db.scheme.bulkCreate(schemes)
+    await db.invoiceNumber.create({ invoiceNumber: paymentRequest.invoiceNumber, originalInvoiceNumber: 'abcdef' })
+    await db.paymentRequest.create(paymentRequest)
+  })
+
+  afterEach(async () => {
+    await db.sequelize.truncate({
+      cascade: true,
+      restartIdentity: true
+    })
+  })
+
+  afterAll(async () => {
+    await db.sequelize.close()
+  })
+
+  test('should save entry into settlement table with invoiceNumber of settlement.invoiceNumber', async () => {
+    await processReturnSettlement(settlement)
+
+    const result = await db.settlement.findOne({ where: { invoiceNumber: settlement.invoiceNumber } })
+    expect(result).not.toBeNull()
+  })
+
+  test('should save one entry into settlement table with invoiceNumber of settlement.invoiceNumber', async () => {
+    await processReturnSettlement(settlement)
+
+    const result = await db.settlement.count({ where: { invoiceNumber: settlement.invoiceNumber } })
+    expect(result).toBe(1)
+  })
+
+  test('should save one entry into settlement table with frn of settlement.frn', async () => {
+    await processReturnSettlement(settlement)
+
+    const result = await db.settlement.count({ where: { frn: settlement.frn } })
```

Comment:
So we are still wanting to get that record by the same mechanism we did to ensure they were only 1, i.e. by `invoiceNumber`